src/main/kotlin/de/craftan/engine/map/graph/Node.kt

```diff
@@ -0,0 +1,132 @@
+package de.craftan.engine.map.graph
+
+import de.craftan.engine.CraftanStructure
+import de.craftan.engine.map.GameTile
+import de.craftan.engine.map.TileDirection
+
+/**
+ * A Node is practically the place where the players can build settlements and cities.
+ * It is the intersection of up to 3 tiles and is connected to other nodes via the edges connected to it
+ */
+data class Node(
+    val structureInfo: StructureInfo,
+    val tiles: MutableSet<GameTile> = mutableSetOf(),
+    val edges: MutableSet<Edge> = mutableSetOf(),
+)
+
+/**
+ * Models a structure placed on an edge or a node.
+ */
+data class StructureInfo(
+    var structure: CraftanStructure? = null,
+)
+
+/**
+ * The direction on a tile a node can be in.
+ * Contains the direction of the tiles the node also borders if they exist.
+ */
+enum class NodeDirection(
+    val otherTileDirections: Pair<TileDirection, TileDirection>,
+) {
+    NORTH_WEST(Pair(TileDirection.WEST, TileDirection.NORTH_WEST)),
+    NORTH(Pair(TileDirection.NORTH_WEST, TileDirection.NORTH_EAST)),
+    NORTH_EAST(Pair(TileDirection.EAST, TileDirection.NORTH_EAST)),
+    SOUTH_EAST(Pair(TileDirection.EAST, TileDirection.SOUTH_EAST)),
+    SOUTH(Pair(TileDirection.SOUTH_WEST, TileDirection.SOUTH_EAST)),
+    SOUTH_WEST(Pair(TileDirection.WEST, TileDirection.SOUTH_WEST)),
+}
+
+/**
+ * Generates empty nodes for all tiles
+ */
+fun generateNodes(tiles: List<GameTile>) {
```

Comment:
I personally would move all of these functions, also including all the edge stuff etc. into a separate file, to have mor clearance of the code. The file could simply be named NodeUtils.kt for example